Add an "in memory" option so non-production deployments can still run

stephen-soltesz opened the pull request:
This change adds an in-memory option to the alertmanager github receiver.

With this change, the service can `-enable-inmemory` instead of depending on the github API. This will be helpful for supporting deployments in staging (to test the receiver behavior) without creating issues in production.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/14)
<!-- Reviewable:end -->

gfr10598 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r1.
Review status: :shipit: complete! 1 of 1 LGTMs obtained

---

*[memory/memory.go, line 17 at r1](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/14#-LFY796G1b2gQAPMM4In:-LFY796HD9B8n6LWpH6a:b-3c2qkg) ([raw file](https://github.com/m-lab/alertmanager-github-receiver/blob/af597c4e2676a8f96371ca34fe0e2734d7d75c6b/memory/memory.go#L17)):*
> ```Go
> 
> // Package memory provides in memory operations on GitHub issues.
> package memory
> ```

Nit:  Not crazy about this package name.  Maybe "fake"?

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/14#-:-LFY7LBo-P3KO-BwT_lj:bnfp4nl)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Review status: :shipit: complete! 1 of 1 LGTMs obtained

---

*[memory/memory.go, line 17 at r1](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/14#-LFY796G1b2gQAPMM4In:-LFYTL-CC57dr_BTO1IC:b-uw81y6) ([raw file](https://github.com/m-lab/alertmanager-github-receiver/blob/af597c4e2676a8f96371ca34fe0e2734d7d75c6b/memory/memory.go#L17)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Nit:  Not crazy about this package name.  Maybe "fake"?
</blockquote></details>

I don't love the name either... it was shorter than some of the alternatives I considered. "fake" reminds me too much of unit testing. 

* `inmemory`
* `issues_memory`
* `issues_noop`
* `issues_local`
* `local`

`local.Client`?

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/14)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Review status: :shipit: complete! 1 of 1 LGTMs obtained

---

*[memory/memory.go, line 17 at r1](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/14#-LFY796G1b2gQAPMM4In:-LF_GRWO0w3wsq1qPm_f:bar0itd) ([raw file](https://github.com/m-lab/alertmanager-github-receiver/blob/af597c4e2676a8f96371ca34fe0e2734d7d75c6b/memory/memory.go#L17)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

I don't love the name either... it was shorter than some of the alternatives I considered. "fake" reminds me too much of unit testing. 

* `inmemory`
* `issues_memory`
* `issues_noop`
* `issues_local`
* `local`

`local.Client`?
</blockquote></details>

I've renamed the 'memory' package to 'local' and moved it to a subdirectory of 'issues'. This groups the 'local' package within the 'issues' package in a way that I think helps explain it. Kind of like "net/http" and "net/http/httptest" .. even though this isn't for testing.

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/14)*
<!-- Sent from Reviewable.io -->
